### PR TITLE
Documentation improvements

### PR DESCRIPTION
**Pull request by malud**
As the product grows with features the documentation and reference needs more structure to find the right things more easily.

This is the first step, split up more generic documentation and examples from the configuration reference (will be generated later on).

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -129,7 +69,7 @@ changes the working directory to `/opt/couper` and loads `my_conf.hcl`.
 
 Couper's configuration file consists of nested configuration blocks that configure
 Web serving and routing of the gateway. Access control is controlled by an
-[Access Control](#access-control) attribute that can be set for many blocks.
+[Access Control](./REFERENCE.md#access-control) attribute that can be set for many blocks.
```
Comment (alex-schneider) [inline docs/README.md:72]: The test should also check such links, too.

File: docs/README.md [hunk 2]
```diff
@@ -1,122 +1,50 @@
...
-| Client(s)          | Browser, App or API Client that sends requests to Couper. |
-| Web Serving        | Couper supports file serving and Web serving for SPA assets. |
-| API                | Configuration block that bundles endpoints under a certain base path. |
-| Access Control     | Couper handles access control for incoming client requests and outgoing backend requests. |
-| Endpoint           | Configuration block that specifies how (and if) requests are sent to backend service(s) after they reach Couper. |
-| Backend            | Configuration block that specifies the connection to a local/remote backend service. |
-| Logging            | Couper provides standard logs for analysis and monitoring. |
-| Backend Service(s) | External API or micro services where Couper fetches data from. |
-| Validation         | Couper supports validation of outgoing and incoming requests to and from the origin. |
+    * [Variables](#variables)
```
Comment (malud) [inline docs/README.md:8]: Variables and Functions are missing in Reference File?

File: docs/README.md [hunk 3]
```diff
@@ -179,171 +103,67 @@ settings {
...
-|:-------------------|:------------|
-| `status`           | HTTP status code |
-| `headers.<name>`   | HTTP response header value for requested lower-case key |
-| `cookies.<name>`   | Value from `Set-Cookie` response header for requested key (&#9888; last wins!) |
-| `body`             | The response message body |
-| `json_body.<name>` | Access json decoded object properties. Media type must be `application/json` or `application/*+json`. |
 
-##### Variable Example
+
+<!--
```
Comment (malud) [inline docs/README.md:135]: The following examples should be at least combined with an Examples listing.

Comment (malud) [reply docs/README.md:135]: Regenerate table-of-contents

File: docs/REFERENCE.md [hunk 4]
```diff
@@ -0,0 +1,651 @@
...
+| *context*                            | Root of the configuration file. |
+| *label*                              | &#9888; Mandatory. |
+| **Nested blocks**                    | **Description** |
+| [CORS Block](#cors-block)            | Configures CORS behavior for [Files Block](#files-block), [SPA Block](#spa-block) and [API Block(s)](#api-block) contexts. |
+| [Files Block](#files-block)          | Configures the file serving. |
+| [SPA Block](#spa-block)              | Configures the Web serving for SPA assets. |
+| [API Block(s)](#api-block)           | Configures routing and communication with backend(s). |
+| [Endpoint Block(s)](#endpoint-block) | Configures specific endpoint(s) for current `Server Block` context. |
+| **Attributes**                       | **Description** |
+| `base_path`                          | <ul><li>Optional.</li><li>Configures the path prefix for all requests.</li><li>*Example:* `base_path = "/api"`</li><li>&#9888; Inherited by nested blocks.</li></ul> |
```
Comment (malud) [inline docs/REFERENCE.md:59]: Should be applied to all html/sub-listings within this document: Replace with proper parent table configuration eg. (block/name, default, required, description).

**Pull request merged**